main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"crypto/rsa"
+	"github.com/joho/godotenv"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"github.com/ufcg-lsd/arrebol-pb-worker/worker"
+	"log"
+	"os"
+)
+
+const (
+	ConfFilePathKey = "CONF_FILE"
+	ServerEndpointKey = "SERVER_ENDPOINT"
+)
+
+func setup(serverEndPoint string, workerId string) {
+	// The setup routine is responsible for generates the rsa key pairs
+	// of the worker and to send the public part to the server.
+	log.Println("Starting to gen rsa key pair with workerid: " + workerId)
+
+	utils.GenAccessKeys(workerId)
+
+	log.Println("Sending pub key to the server")
+	url := serverEndPoint + "/workers/publicKey"
+	requestBody := &map[string]*rsa.PublicKey{"key": utils.GetPublicKey(workerId)}
+	httpResp := utils.Post(requestBody, url)
+
+	if httpResp.StatusCode != 201 {
+		log.Fatal("Unable to send public key to the server")
+	}
+}
+
+func isTokenValid(token string) bool {
+	return true
+}
+
+func main() {
+	// this main function start the worker following the chosen implementation
+	// passed by arg in the cli. The defaultWorker is started if no arg has been received.
+	err := godotenv.Load()
+
+	if err != nil {
+		log.Println("No .env file found")
+	}
+
+	switch len(os.Args) {
+	case 2:
```

Comment:
didn't  understand why do need this case 2 here

Reply:
I'll add a comment explaining this. It is the way the system choose between workers' implementation. When the main is executed passing a command line parameter, the systems understand that the second param is the worker implementation...

Reply:
do you already have a second possible worker implementation? If not, remove this.

Reply:
done